fix: fixes issue with metrics process collector

bilbercode opened the pull request:
<!--
Thank you for contributing to CoreDNS!
Please provide the following information to help us make the most of your pull request:
-->

### 1. Why is this pull request needed and what does it do?
fixes call to create NewProcessCollector in metrics, ref https://github.com/prometheus/client_golang/commit/7858729281ec582767b20e0d696b6041d995d5e0
### 2. Which issues (if any) are related?
GitHub issue #2090
### 3. Which documentation changes (if any) need to be made?
No changes required to documentation

miekg commented:
Fails to build with:
~~~
plugin/metrics/metrics.go:42:49: not enough arguments in call to prometheus.NewProcessCollector
plugin/metrics/metrics.go:42:50: undefined: prometheus.ProcessCollectorOpts
~~~

fastest963 commented:
I don't understand why the build is failing. The latest github.com/prometheus/client_golang has that struct and I don't see us vendoring that. @miekg any ideas?

miekg commented:
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] fix: fixes is..." ]
>I don't understand why the build is failing. The latest github.com/prometheus/client_golang has that struct and I don't see us vendoring that. @miekg any ideas?

We pull down a specific version, see the `Makefile`, might be worth bumping that 
version.

(Unrelated we should probably bump everything in vendor as well).

miekg commented:
hmmm, now locally *without* this PR my test also doesn't compile, uh?
(doing something else [code wise], but I'll take a look later)

fastest963 commented:
@miekg how do you feel about v0.9.0-pre1 locking for now? The README talks about their philosophy: https://github.com/prometheus/client_golang/blob/master/README.md

miekg commented:
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] fix: fixes is..." ]
>@miekg how do you feel about v0.9.0-pre1 locking for now? The README talks about their philosophy: https://github.com/prometheus/client_golang/blob/master/README.md

We use 'git checkout -q v0.8.0' now, is v0.9.0-pre1 worse/better? Can't tell, 
we're also not the most advanced users, so if you think is OK, I'm fine with 
that.

/Miek

--
Miek Gieben

bilbercode commented:
Hey guys, my bad on this PR. I did a `go get -u ./...` which pulled from master on Prometheus. Obviously they are doing some work on the process collector

fastest963 commented:
I updated to v0.9.0-pre1 but is that not new enough for the change @bilbercode? Looks like the travis build still failed saying the method signature is still the old one.

@bilbercode is it better if we just wait until v0.9.0 is tagged (and complete) before merging this change?

bilbercode commented:
I would stick with 0.8.0 for now and not roll to the pre release, the interface could change :)

fastest963 commented:
Sounds good. We'll punt on this until v0.9.0 is complete.

bilbercode commented:
I'm actively developing an in house plugin and regularly use prometheus (prom) for instrumentation, if a newer version is released I will try and fix any issues an update this PR 👍

hmage commented:
v0.9.0 was just released -- https://github.com/prometheus/client_golang/releases/tag/v0.9.0

Guess it's time to bump the version.

miekg commented:
yes, please rebase and do all the magic to make it pass tests.

bilbercode commented:
Hey folks
I rebased from master and updated the version to the latest prom, most tests pass, however this line in the benchmark block of the `Makefile`
https://github.com/coredns/coredns/blob/d9efa9637460e799afa065c771ff009057e2bf85/Makefile#L70 will mostly fail if there are any code changes in the vendor deps that were patching because were checking out old code against a newer version of the deps

yongtang commented:
Guess we could ignore the benchmark failure? If there is an issue after the merge we could revert I think.

bilbercode commented:
I agree, I don't think we should loose the benchmark comparison, when were dealing with issues like this is mostly an edge case :)

miekg commented:
This needs a rebase?

miekg commented:
want to rebase and see this still applies?
